In [1]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt

### Import the all_players.csv

using Pandas

In [2]:
# file directory
file_dir = "C:/Users/mamoo/UT_DataScience/UTA-VIRT-DATA-PT-02-2021-U-B/FinalProject/Resources/"

# import the all_player.csv
all_players_df = pd.read_csv(f"{file_dir}all_players_2004to2020.csv")
pd.set_option("display.max_columns", None) # To show all columns explicitly
all_players_df

Player Club  POS  GP  GS  MINS  G  A  SHTS  SOG  GWG PKG/A  \
0     Alecko Eskandarian  NaN    F   4   4   365  4  1    15    8    0   0/0   
1           Jaime Moreno  NaN    F   4   4   389  2  1     5    2    1   0/0   
2            Davy Arnaud  NaN    M   4   4   360  2  1     5    2    1   0/0   
3        Taylor Twellman  NaN    F   3   3   300  2  0    11    5    0   0/0   
4         Earnie Stewart  NaN    M   4   4   321  1  2     1    1    1   0/0   
...                  ...  ...  ...  ..  ..   ... .. ..   ...  ...  ...   ...   
4966   Alejandro Pozuelo  NaN    M   1   1   120  0  0     5    0    0   0/0   
4967         Franco Jara  NaN    F   2   2   210  0  0     5    1    0   0/0   
4968           Andy Polo  NaN    M   1   1   105  0  0     6    3    0   0/0   
4969     Cristian Roldan  NaN    M   4   4   360  0  0     6    1    0   0/0   
4970        Joevin Jones  NaN  D-M   4   4   260  0  0     8    1    0   0/0   

      HmG  RdG  G/90min    SC%  GWA  HmA  RdA  A/90min  FC  FS  OFF  YC  RC  \
0       1    3     0.99   26.7    0    1    0     0.25   9   8    3   1   0   
1       2    0     0.46   40.0    0    0    1     0.23  10  14    0   0   0   
2       2    0     0.50   40.0    1    1    0     0.25  11  11    4   1   0   
3       0    2     0.60   18.2    0    0    0     0.00   4  11    7   0   0   
4       0    1     0.28  100.0    0    2    0     0.56   4   2    4   1   0   
...   ...  ...      ...    ...  ...  ...  ...      ...  ..  ..  ...  ..  ..   
4966    0    0     0.00    0.0    0    0    0     0.00   2   0    0   0   0   
4967    0    0     0.00    0.0    0    0    0     0.00   6   2    3   0   0   
4968    0    0     0.00    0.0    0    0    0     0.00   0   3    0   0   0   
4969    0    0     0.00    0.0    0    0    0     0.00   5  11    0   0   0   
4970    0    0     0.00    0.0    0    0    0     0.00   1   7    0   0   0   

        SOG%  Year Season  
0      53.33  2004   post  
1      40.00  2004   post  
2      40.00  2004   post  
3      45.45  2004   post  
4     100.00  2004   post  
...      ...   ...    ...  
4966    0.00  2020   post  
4967   20.00  2020   post  
4968   50.00  2020   post  
4969   16.67  2020   post  
4970   12.50  2020   post  

[4971 rows x 28 columns]

In [3]:
# Check all the column headers.
all_players_df.columns

Index(['Player', 'Club', 'POS', 'GP', 'GS', 'MINS', 'G', 'A', 'SHTS', 'SOG',
       'GWG', 'PKG/A', 'HmG', 'RdG', 'G/90min', 'SC%', 'GWA', 'HmA', 'RdA',
       'A/90min', 'FC', 'FS', 'OFF', 'YC', 'RC', 'SOG%', 'Year', 'Season'],
      dtype='object')

Most of the column headers are in abbreviated form which is hard to understand. We need to rename columns with a little elaborate titles.

In [4]:
all_players_df = all_players_df.rename(columns={"POS": "Position", 
                               "GP": "GamesPlayed",
                              "GS": "GamesStarted",
                              "MINS": "MinutesPlayed",
                              "G": "GoalsScored",
                              "A": "Assists",
                              "SHTS": "Shots",
                              "SOG": "ShotsOnGoal",
                              "GWG": "GameWinningGoal",
                              "PKG/A": "PenaltyKicks",
                              "HmG": "HomeGoals",
                              "RdG": "RoadGoals",
                              "G/90min": "GoalsPer90Minutes",
                              "SC%": "ShotConversion",
                              "GWA": "GameWinningAssists",
                              "HmA": "HomeAssists",
                              "RdA": "RoadAssists",
                              "A/90min": "AssistsPer90Min",
                              "FC": "FoulsCommitted",
                              "FS": "FoulsSufferred",
                              "OFF": "Offsides",
                              "YC": "YellowCards",
                              "RC": "RedCards",
                              "SOG%": "ShotsOnGoalPercent",
                              "Year": "Year",
                              })
all_players_df.head(2)

Player Club Position  GamesPlayed  GamesStarted  MinutesPlayed  \
0  Alecko Eskandarian  NaN        F            4             4            365   
1        Jaime Moreno  NaN        F            4             4            389   

   GoalsScored  Assists  Shots  ShotsOnGoal  GameWinningGoal PenaltyKicks  \
0            4        1     15            8                0          0/0   
1            2        1      5            2                1          0/0   

   HomeGoals  RoadGoals  GoalsPer90Minutes  ShotConversion  \
0          1          3               0.99            26.7   
1          2          0               0.46            40.0   

   GameWinningAssists  HomeAssists  RoadAssists  AssistsPer90Min  \
0                   0            1            0             0.25   
1                   0            0            1             0.23   

   FoulsCommitted  FoulsSufferred  Offsides  YellowCards  RedCards  \
0               9               8         3            1         0   
1              10              14         0            0         0   

   ShotsOnGoalPercent  Year Season  
0               53.33  2004   post  
1               40.00  2004   post

In [5]:
# Show any rows where 'Club' title is empty
all_players_df[(all_players_df['Club'] == '') | (all_players_df['Club'].isnull())]

Player Club Position  GamesPlayed  GamesStarted  \
0     Alecko Eskandarian  NaN        F            4             4   
1           Jaime Moreno  NaN        F            4             4   
2            Davy Arnaud  NaN        M            4             4   
3        Taylor Twellman  NaN        F            3             3   
4         Earnie Stewart  NaN        M            4             4   
...                  ...  ...      ...          ...           ...   
4966   Alejandro Pozuelo  NaN        M            1             1   
4967         Franco Jara  NaN        F            2             2   
4968           Andy Polo  NaN        M            1             1   
4969     Cristian Roldan  NaN        M            4             4   
4970        Joevin Jones  NaN      D-M            4             4   

      MinutesPlayed  GoalsScored  Assists  Shots  ShotsOnGoal  \
0               365            4        1     15            8   
1               389            2        1      5            2   
2               360            2        1      5            2   
3               300            2        0     11            5   
4               321            1        2      1            1   
...             ...          ...      ...    ...          ...   
4966            120            0        0      5            0   
4967            210            0        0      5            1   
4968            105            0        0      6            3   
4969            360            0        0      6            1   
4970            260            0        0      8            1   

      GameWinningGoal PenaltyKicks  HomeGoals  RoadGoals  GoalsPer90Minutes  \
0                   0          0/0          1          3               0.99   
1                   1          0/0          2          0               0.46   
2                   1          0/0          2          0               0.50   
3                   0          0/0          0          2               0.60   
4                   1          0/0          0          1               0.28   
...               ...          ...        ...        ...                ...   
4966                0          0/0          0          0               0.00   
4967                0          0/0          0          0               0.00   
4968                0          0/0          0          0               0.00   
4969                0          0/0          0          0               0.00   
4970                0          0/0          0          0               0.00   

      ShotConversion  GameWinningAssists  HomeAssists  RoadAssists  \
0               26.7                   0            1            0   
1               40.0                   0            0            1   
2               40.0                   1            1            0   
3               18.2                   0            0            0   
4              100.0                   0            2            0   
...              ...                 ...          ...          ...   
4966             0.0                   0            0            0   
4967             0.0                   0            0            0   
4968             0.0                   0            0            0   
4969             0.0                   0            0            0   
4970             0.0                   0            0            0   

      AssistsPer90Min  FoulsCommitted  FoulsSufferred  Offsides  YellowCards  \
0                0.25               9               8         3            1   
1                0.23              10              14         0            0   
2                0.25              11              11         4            1   
3                0.00               4              11         7            0   
4                0.56               4               2         4            1   
...               ...             ...             ...       ...          ...   
4966             0.00               2               0     

So there are 4280 rows out of 4971 rows where the *Club* column has no value. In our analysis, the "Club" information is crucial. So we may drop the rows where there is no info for *Club*. 

In [6]:
all_players_df[(all_players_df['Club'].isnull())]

Player Club Position  GamesPlayed  GamesStarted  \
0     Alecko Eskandarian  NaN        F            4             4   
1           Jaime Moreno  NaN        F            4             4   
2            Davy Arnaud  NaN        M            4             4   
3        Taylor Twellman  NaN        F            3             3   
4         Earnie Stewart  NaN        M            4             4   
...                  ...  ...      ...          ...           ...   
4966   Alejandro Pozuelo  NaN        M            1             1   
4967         Franco Jara  NaN        F            2             2   
4968           Andy Polo  NaN        M            1             1   
4969     Cristian Roldan  NaN        M            4             4   
4970        Joevin Jones  NaN      D-M            4             4   

      MinutesPlayed  GoalsScored  Assists  Shots  ShotsOnGoal  \
0               365            4        1     15            8   
1               389            2        1      5            2   
2               360            2        1      5            2   
3               300            2        0     11            5   
4               321            1        2      1            1   
...             ...          ...      ...    ...          ...   
4966            120            0        0      5            0   
4967            210            0        0      5            1   
4968            105            0        0      6            3   
4969            360            0        0      6            1   
4970            260            0        0      8            1   

      GameWinningGoal PenaltyKicks  HomeGoals  RoadGoals  GoalsPer90Minutes  \
0                   0          0/0          1          3               0.99   
1                   1          0/0          2          0               0.46   
2                   1          0/0          2          0               0.50   
3                   0          0/0          0          2               0.60   
4                   1          0/0          0          1               0.28   
...               ...          ...        ...        ...                ...   
4966                0          0/0          0          0               0.00   
4967                0          0/0          0          0               0.00   
4968                0          0/0          0          0               0.00   
4969                0          0/0          0          0               0.00   
4970                0          0/0          0          0               0.00   

      ShotConversion  GameWinningAssists  HomeAssists  RoadAssists  \
0               26.7                   0            1            0   
1               40.0                   0            0            1   
2               40.0                   1            1            0   
3               18.2                   0            0            0   
4              100.0                   0            2            0   
...              ...                 ...          ...          ...   
4966             0.0                   0            0            0   
4967             0.0                   0            0            0   
4968             0.0                   0            0            0   
4969             0.0                   0            0            0   
4970             0.0                   0            0            0   

      AssistsPer90Min  FoulsCommitted  FoulsSufferred  Offsides  YellowCards  \
0                0.25               9               8         3            1   
1                0.23              10              14         0            0   
2                0.25              11              11         4            1   
3                0.00               4              11         7            0   
4                0.56               4               2         4            1   
...               ...             ...             ...       ...          ...   
4966             0.00               2               0     

In [7]:
all_players_df = all_players_df.drop(all_players_df[all_players_df['Club'].isnull()].index)
all_players_df

Player  Club Position  GamesPlayed  GamesStarted  \
3814       Diego Rossi  LAFC        F           19            19   
3815      Gyasi Zardes   CLB        F           21            20   
3816     Raul Ruidiaz    SEA        F           17            17   
3817      Robert Beric   CHI        F           23            22   
3818     Jordan Morris   SEA        F           22            18   
...                ...   ...      ...          ...           ...   
4500  Sebastian Mendez   ORL        M           19            11   
4501        Kelyn Rowe    NE        M           16            10   
4502      Siem de Jong   CIN        M           15             8   
4503     Maxime Chanot   NYC        D           20            20   
4504     Nicolas Figal   MIA        D           21            21   

      MinutesPlayed  GoalsScored  Assists  Shots  ShotsOnGoal  \
3814           1698           14        4     71           34   
3815           1717           12        4     41           19   
3816           1427           12        4     55           23   
3817           1931           12        1     62           29   
3818           1724           10        8     39           18   
...             ...          ...      ...    ...          ...   
4500           1087            0        0     17            3   
4501            896            0        0     17            4   
4502            793            0        0     18            5   
4503           1712            0        0     19            4   
4504           1746            0        0     22            9   

      GameWinningGoal PenaltyKicks  HomeGoals  RoadGoals  GoalsPer90Minutes  \
3814                1        1-Jan         11          3               0.74   
3815                2          0/2          8          4               0.63   
3816                4        1-Jan          8          4               0.76   
3817                1        2-Jan          7          5               0.56   
3818                3          0/0          8          2               0.52   
...               ...          ...        ...        ...                ...   
4500                0          0/0          0          0               0.00   
4501                0          0/0          0          0               0.00   
4502                0          0/1          0          0               0.00   
4503                0          0/0          0          0               0.00   
4504                0          0/0          0          0               0.00   

      ShotConversion  GameWinningAssists  HomeAssists  RoadAssists  \
3814            19.7                   2            3            1   
3815            29.3                   1            2            2   
3816            21.8                   0            2            2   
3817            19.4                   0            1            0   
3818            25.6                   0            5            3   
...              ...                 ...          ...          ...   
4500             0.0                   0            0            0   
4501             0.0                   0            0            0   
4502             0.0                   0            0            0   
4503             0.0                   0            0            0   
4504             0.0                   0            0            0   

      AssistsPer90Min  FoulsCommitted  FoulsSufferred  Offsides  YellowCards  \
3814             0.21               6              20        12            0   
3815             0.21              19              27        17            1   
3816             0.25              11              13        13            2   
3817             0.05              38              58        28            1   
3818             0.42               6              12        13            1   
...               ...             ...             ...       ...          ...   
4500             0.00              32               9         1     

In [8]:
all_players_df['Club'].nunique()

29

In [9]:
all_players_df['Club'].unique()

array(['LAFC', 'CLB', 'SEA', 'CHI', 'ORL', 'LA', 'TOR', 'MIN', 'POR',
       'MTL', 'PHI', 'NE', 'RSL', 'HOU', 'SJ', 'DAL', 'SKC', 'NYC', 'VAN',
       'MIA', 'COL', 'RBNY', 'NSH', 'ATL', 'DC', 'CIN', 'LFC', 'MCF',
       'NY'], dtype=object)

In [10]:
all_players_df.isnull().sum()

Player                  0
Club                    0
Position                0
GamesPlayed             0
GamesStarted            0
MinutesPlayed           0
GoalsScored             0
Assists                 0
Shots                   0
ShotsOnGoal             0
GameWinningGoal         0
PenaltyKicks            0
HomeGoals               0
RoadGoals               0
GoalsPer90Minutes       0
ShotConversion          0
GameWinningAssists      0
HomeAssists             0
RoadAssists             0
AssistsPer90Min         0
FoulsCommitted          0
FoulsSufferred          0
Offsides                0
YellowCards             0
RedCards                0
ShotsOnGoalPercent    151
Year                    0
Season                  0
dtype: int64

It looks like the **ShotsOnGoalPercent** column has 151 null values (out of 691 rows).

The values under *ShotsOnGoalPercent* = *ShotsOnGoal* / *Shots*. Since this column is derived from mathematical manipulation of other two columns, if we drop this column the information should still be there in the dataframe.

In [11]:
# Drop the column "ShotsOnGoalPercent"
all_players_df = all_players_df.drop("ShotsOnGoalPercent", axis='columns')
all_players_df.head(2)

Player  Club Position  GamesPlayed  GamesStarted  MinutesPlayed  \
3814   Diego Rossi  LAFC        F           19            19           1698   
3815  Gyasi Zardes   CLB        F           21            20           1717   

      GoalsScored  Assists  Shots  ShotsOnGoal  GameWinningGoal PenaltyKicks  \
3814           14        4     71           34                1        1-Jan   
3815           12        4     41           19                2          0/2   

      HomeGoals  RoadGoals  GoalsPer90Minutes  ShotConversion  \
3814         11          3               0.74            19.7   
3815          8          4               0.63            29.3   

      GameWinningAssists  HomeAssists  RoadAssists  AssistsPer90Min  \
3814                   2            3            1             0.21   
3815                   1            2            2             0.21   

      FoulsCommitted  FoulsSufferred  Offsides  YellowCards  RedCards  Year  \
3814               6              20        12            0         0  2020   
3815              19              27        17            1         0  2020   

     Season  
3814    reg  
3815    reg

In [12]:
# The "PenaltyKicks" column has also some inconsistent data. We're focused on the team or club.
# As we still have 28 columns of data, we decide to to drop this column.
all_players_df = all_players_df.drop("PenaltyKicks", axis='columns')
all_players_df.head(2)

Player  Club Position  GamesPlayed  GamesStarted  MinutesPlayed  \
3814   Diego Rossi  LAFC        F           19            19           1698   
3815  Gyasi Zardes   CLB        F           21            20           1717   

      GoalsScored  Assists  Shots  ShotsOnGoal  GameWinningGoal  HomeGoals  \
3814           14        4     71           34                1         11   
3815           12        4     41           19                2          8   

      RoadGoals  GoalsPer90Minutes  ShotConversion  GameWinningAssists  \
3814          3               0.74            19.7                   2   
3815          4               0.63            29.3                   1   

      HomeAssists  RoadAssists  AssistsPer90Min  FoulsCommitted  \
3814            3            1             0.21               6   
3815            2            2             0.21              19   

      FoulsSufferred  Offsides  YellowCards  RedCards  Year Season  
3814              20        12            0         0  2020    reg  
3815              27        17            1         0  2020    reg

In [13]:
all_players_df.describe()

GamesPlayed  GamesStarted  MinutesPlayed  GoalsScored     Assists  \
count   691.000000    691.000000     691.000000   691.000000  691.000000   
mean     11.780029      8.451520     758.933430     1.170767    1.134588   
std       7.579966      7.172525     620.201766     2.071144    1.767043   
min       0.000000      0.000000       0.000000     0.000000    0.000000   
25%       5.000000      1.000000     164.000000     0.000000    0.000000   
50%      13.000000      8.000000     705.000000     0.000000    0.000000   
75%      19.000000     15.000000    1277.000000     2.000000    2.000000   
max      23.000000     23.000000    2070.000000    14.000000   10.000000   

            Shots  ShotsOnGoal  GameWinningGoal   HomeGoals   RoadGoals  \
count  691.000000   691.000000       691.000000  691.000000  691.000000   
mean    10.479016     3.772793         0.316932    0.671491    0.499276   
std     12.959046     5.435168         0.735824    1.360109    0.962146   
min      0.000000     0.000000         0.000000    0.000000    0.000000   
25%      1.000000     0.000000         0.000000    0.000000    0.000000   
50%      6.000000     1.000000         0.000000    0.000000    0.000000   
75%     16.000000     5.000000         0.000000    1.000000    1.000000   
max     75.000000    34.000000         5.000000   11.000000    6.000000   

       GoalsPer90Minutes  ShotConversion  GameWinningAssists  HomeAssists  \
count         691.000000      691.000000          691.000000   691.000000   
mean            0.134486        7.394211            0.312590     0.633864   
std             0.397531       13.141043            0.700394     1.157438   
min             0.000000        0.000000            0.000000     0.000000   
25%             0.000000        0.000000            0.000000     0.000000   
50%             0.000000        0.000000            0.000000     0.000000   
75%             0.160000       12.300000            0.000000     1.000000   
max             7.500000      100.000000            6.000000     7.000000   

       RoadAssists  AssistsPer90Min  FoulsCommitted  FoulsSufferred  \
count   691.000000       691.000000      691.000000      691.000000   
mean      0.500724         0.111520       10.581766        9.765557   
std       0.960638         0.261551        9.867244       10.854907   
min       0.000000         0.000000        0.000000        0.000000   
25%       0.000000         0.000000        2.000000        1.000000   
50%       0.000000         0.000000        8.000000        6.000000   
75%       1.000000         0.170000       16.000000       14.000000   
max       7.000000         4.740000       55.000000       58.000000   

         Offsides  YellowCards    RedCards    Year  
count  691.000000   691.000000  691.000000   691.0  
mean     1.429812     1.519537    0.066570  2020.0  
std      2.990649     1.848809    0.260817     0.0  
min      0.000000     0.000000    0.000000  2020.0  
25%      0.000000     0.000000    0.000000  2020.0  
50%      0.000000     1.000000    0.000000  2020.0  
75%      1.000000     2.000000    0.000000  2020.0  
max     28.000000     9.000000    2.000000  2020.0

In [14]:
all_players_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 691 entries, 3814 to 4504
Data columns (total 26 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Player              691 non-null    object 
 1   Club                691 non-null    object 
 2   Position            691 non-null    object 
 3   GamesPlayed         691 non-null    int64  
 4   GamesStarted        691 non-null    int64  
 5   MinutesPlayed       691 non-null    int64  
 6   GoalsScored         691 non-null    int64  
 7   Assists             691 non-null    int64  
 8   Shots               691 non-null    int64  
 9   ShotsOnGoal         691 non-null    int64  
 10  GameWinningGoal     691 non-null    int64  
 11  HomeGoals           691 non-null    int64  
 12  RoadGoals           691 non-null    int64  
 13  GoalsPer90Minutes   691 non-null    float64
 14  ShotConversion      691 non-null    float64
 15  GameWinningAssists  691 non-null    int64  
 16  Home

The **all_players_df** is now **cleaned** with total of **26 columns, 691 rows** of non-null data with consistent data types. 

### Export the clean df to .csv

In [15]:
# Rename the 'Club' column's club abbreviations.
all_players_cleaned = all_players_df.replace({'Club': {'LAFC':'Los Angeles FC', 
                                                       'CLB':'Columbus Crew', 
                                                       'SEA':'Seattle Sounders FC', 
                                                       'CHI':'Chicago FIre FC',
                                                       'ORL':'Orlando City SC',
                                                       'LA':'LA Galaxy',
                                                       'TOR':'Toronto FC',
                                                       'MIN':'Minnesota United FC',
                                                       'POR':'Portland Timbers',
                                                       'MTL':'CF Montreal',
                                                       'PHI':'Philadelphia Union',
                                                       'NE':'New England Revolution',
                                                       'RSL':'Real Salt Lake',
                                                       'HOU':'Houston Dynamo FC',
                                                       'SJ':'San Jose Earthquakes',
                                                       'DAL':'FC Dallas',
                                                       'SKC':'Sporting Kansas City',
                                                       'NYC':'New York City',
                                                       'VAN':'Vancouver Whitecaps FC',
                                                       'MIA':'Inter Miami CF',
                                                       'COL':'Colorado Rapids',
                                                       'RBNY':'New York City Red Bulls',
                                                       'NSH':'Nashville SC',
                                                       'ATL':'Atlanta United', 
                                                       'DC':'D.C. United',
                                                       'CIN':'FC Cincinnati',
                                                       'LFC':'Los Angeles FC', 
                                                       'MCF':'Inter Miami CF',
                                                       'NY': 'New York City'}})
all_players_cleaned.tail()

Player                    Club Position  GamesPlayed  \
4500  Sebastian Mendez         Orlando City SC        M           19   
4501        Kelyn Rowe  New England Revolution        M           16   
4502      Siem de Jong           FC Cincinnati        M           15   
4503     Maxime Chanot           New York City        D           20   
4504     Nicolas Figal          Inter Miami CF        D           21   

      GamesStarted  MinutesPlayed  GoalsScored  Assists  Shots  ShotsOnGoal  \
4500            11           1087            0        0     17            3   
4501            10            896            0        0     17            4   
4502             8            793            0        0     18            5   
4503            20           1712            0        0     19            4   
4504            21           1746            0        0     22            9   

      GameWinningGoal  HomeGoals  RoadGoals  GoalsPer90Minutes  \
4500                0          0          0                0.0   
4501                0          0          0                0.0   
4502                0          0          0                0.0   
4503                0          0          0                0.0   
4504                0          0          0                0.0   

      ShotConversion  GameWinningAssists  HomeAssists  RoadAssists  \
4500             0.0                   0            0            0   
4501             0.0                   0            0            0   
4502             0.0                   0            0            0   
4503             0.0                   0            0            0   
4504             0.0                   0            0            0   

      AssistsPer90Min  FoulsCommitted  FoulsSufferred  Offsides  YellowCards  \
4500              0.0              32               9         1            4   
4501              0.0              13               4         0            3   
4502              0.0               7               8         1            0   
4503              0.0              18               6         1            5   
4504              0.0              22              26         0            4   

      RedCards  Year Season  
4500         1  2020    reg  
4501         0  2020    reg  
4502         0  2020    reg  
4503         1  2020    reg  
4504         0  2020    reg

In [16]:
# Drop the 'Season' column as it doesn't have any significant input to the analysis.
all_players_cleaned = all_players_cleaned.drop("Season", axis='columns')
all_players_cleaned.head(2)

Player            Club Position  GamesPlayed  GamesStarted  \
3814   Diego Rossi  Los Angeles FC        F           19            19   
3815  Gyasi Zardes   Columbus Crew        F           21            20   

      MinutesPlayed  GoalsScored  Assists  Shots  ShotsOnGoal  \
3814           1698           14        4     71           34   
3815           1717           12        4     41           19   

      GameWinningGoal  HomeGoals  RoadGoals  GoalsPer90Minutes  \
3814                1         11          3               0.74   
3815                2          8          4               0.63   

      ShotConversion  GameWinningAssists  HomeAssists  RoadAssists  \
3814            19.7                   2            3            1   
3815            29.3                   1            2            2   

      AssistsPer90Min  FoulsCommitted  FoulsSufferred  Offsides  YellowCards  \
3814             0.21               6              20        12            0   
3815             0.21              19              27        17            1   

      RedCards  Year  
3814         0  2020  
3815         0  2020

In [26]:
# Export the clean dataframe to "all_players_clean.csv"
csv_dir = "C:/Users/mamoo/UT_DataScience/UTA-VIRT-DATA-PT-02-2021-U-B/FinalProject/Resources/Clean_csv/"
all_players_cleaned.to_csv(f"{csv_dir}all_players_cleaned.csv")

In [18]:
all_players_cleaned.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 691 entries, 3814 to 4504
Data columns (total 25 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Player              691 non-null    object 
 1   Club                691 non-null    object 
 2   Position            691 non-null    object 
 3   GamesPlayed         691 non-null    int64  
 4   GamesStarted        691 non-null    int64  
 5   MinutesPlayed       691 non-null    int64  
 6   GoalsScored         691 non-null    int64  
 7   Assists             691 non-null    int64  
 8   Shots               691 non-null    int64  
 9   ShotsOnGoal         691 non-null    int64  
 10  GameWinningGoal     691 non-null    int64  
 11  HomeGoals           691 non-null    int64  
 12  RoadGoals           691 non-null    int64  
 13  GoalsPer90Minutes   691 non-null    float64
 14  ShotConversion      691 non-null    float64
 15  GameWinningAssists  691 non-null    int64  
 16  Home

In [19]:
# Group by "Club" and "Year"
all_players_grouped = all_players_cleaned.groupby(['Club','Year']).sum()
all_players_grouped

GamesPlayed  GamesStarted  MinutesPlayed  \
Club                    Year                                             
Atlanta United          2020          330           230          20634   
CF Montreal             2020          305           236          20996   
Chicago FIre FC         2020          318           230          20646   
Colorado Rapids         2020          243           175          15632   
Columbus Crew           2020          322           230          20725   
D.C. United             2020          320           234          20843   
FC Cincinnati           2020          305           229          20611   
FC Dallas               2020          312           220          19800   
Houston Dynamo FC       2020          318           229          20622   
Inter Miami CF          2020          319           226          20398   
LA Galaxy               2020          298           220          19655   
Los Angeles FC          2020          312           220          19763   
Minnesota United FC     2020          279           209          18968   
Nashville SC            2020          333           232          20887   
New England Revolution  2020          334           232          20814   
New York City           2020          324           230          20658   
New York City Red Bulls 2020          332           230          20599   
Orlando City SC         2020          330           230          20641   
Philadelphia Union      2020          316           230          20700   
Portland Timbers        2020          326           230          20685   
Real Salt Lake          2020          302           220          19747   
San Jose Earthquakes    2020          324           230          20671   
Seattle Sounders FC     2020          313           222          19957   
Sporting Kansas City    2020          283           210          18874   
Toronto FC              2020          328           230          20700   
Vancouver Whitecaps FC  2020          314           226          20197   

                              GoalsScored  Assists  Shots  ShotsOnGoal  \
Club                    Year                                             
Atlanta United          2020           23       26    219           81   
CF Montreal             2020           34       30    257           98   
Chicago FIre FC         2020           33       31    319          111   
Colorado Rapids         2020           27       32    231           90   
Columbus Crew           2020           34       36    251           91   
D.C. United             2020           23       17    224           73   
FC Cincinnati           2020           11        6    233           65   
FC Dallas               2020           28       25    278          104   
Houston Dynamo FC       2020           30       29    324          117   
Inter Miami CF          2020           26       27    295           95   
LA Galaxy               2020           26       23    249           94   
Los Angeles FC          2020           45       43    358          133   
Minnesota United FC     2020           35       37    300          101   
Nashville SC            2020           25       26    278          100   
New England Revolution  2020           25       24    351          111   
New York City           2020           36       28    356          145   
New York City Red Bulls 2020           29       30    244           85   
Orlando City SC         2020           39       39    266          100   
Philadelphia Union      2020           43       43    297          123   
Portland Timbers        2020           46       48    259          107   
Real Salt Lake          2020           23       19    277           78   
San Jose Earthquakes    2020           34       30    305          116   
Seattle Sounders FC     2020           41       46    281          102   
Sporting Kansas City    2020           37       31    313          103   
Toronto FC              2020     

In [25]:
# Export the grouped dataframe to "all_players_grouped.csv"
csv_dir = "C:/Users/mamoo/UT_DataScience/UTA-VIRT-DATA-PT-02-2021-U-B/FinalProject/Resources/Clean_csv/"
all_players_grouped.to_csv(f"{csv_dir}all_players_grouped.csv")